hudi-client/hudi-spark-client/src/test/java/org/apache/hudi/execution/bulkinsert/TestBulkInsertInternalPartitionerForRows.java

```diff
@@ -103,8 +103,7 @@ public void testBulkInsertInternalPartitioner(BulkInsertSortMode sortMode,
                                                 boolean isGloballySorted,
                                                 boolean isLocallySorted,
                                                 boolean populateMetaFields) {
-    Dataset<Row> records1 = generateTestRecords();
-    Dataset<Row> records2 = generateTestRecords();
+    Dataset<Row> records = generateTestRecords();
```

Comment:
not sure why the existing test case runs the same logic twice with `records1` and `records2`.

Reply:
`testCustomColumnSortPartitionerWithRows` was copied from `testBulkInsertInternalPartitioner`. And I looked `org.apache.hudi.execution.bulkinsert.TestBulkInsertInternalPartitioner#testBulkInsertInternalPartitioner:177`,  it actually generates two records sets with different union times:

```java
JavaRDD<HoodieRecord> records1 = generateTestRecordsForBulkInsert(jsc);
    JavaRDD<HoodieRecord> records2 = generateTripleTestRecordsForBulkInsert(jsc);
```

So I think this should be a mistake, and I think union it twice should be enough(Here different union times for different partitions?)

Reply:
There is no change for any codes in the write path, so why the tests run successfully for Spark 3.1 or 2.4 ?

Reply:
the test only passes spark 2.4 which is an coincident. The existing test logic asserts 2 rdd partitions after re-partition by the partitioner. with spark 2.4's sort and coalesce, it gives 2 and passes the test as a local partitioner. The correct expectation is the partitioner is doing global sort and the resulting num partition should be 2 or less, which is what spark 3 gives us.